### ISSUE TITLE
SQS add-permission does not support adding a non-account principal

### ISSUE THREAD

**Issue by majsty** [aws/aws-cli#7416]
### Describe the bug

The [docs for add-permission](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/sqs/add-permission.html) describe the operation as follows:
>Adds a permission to a queue for a specific [principal](https://docs.aws.amazon.com/general/latest/gr/glos-chap.html#P)

The linked definition of _principal_ describes it as:
>The user, service, or account that receives permissions that are defined in a policy

However, the `add-permission` operation only seems to work with an Account ID.

### Expected Behavior

In addition to specifying an Account ID the operation should accept the Arn for _any_ principal (user, role, etc)

### Current Behavior

When specifying an Arn for a user (or a UserId), the CLI returns the following error:

`An error occurred (InvalidParameterValue) when calling the AddPermission operation: Value [Arn] for parameter PrincipalId is invalid. Reason: Unable to verify principal.`

### Reproduction Steps

1. Create a user in IAM and note the Arn
2. Create a SQS queue and note the URL
3. call `aws sqs add-permission --queue-url [URL from Step 2] --label Jared --aws-account-ids [Arn from Step 1] --actions SendMessage`

Expected: it works

### Possible Solution

Perhaps the validation of the `--aws-account-ids` parameter only validates an Account ID, and needs to be expanded to also allow an Arn to be specified?

### Additional Information/Context

If non-account principals are simply not supported by this operation, then the documentation should be updated to emphasise this.

### CLI version used

2.8.2

### Environment details (OS name and version, etc.)

macOS 12.5.1

**Comment by aBurmeseDev** [comment]
Hi @majsty, thanks for reaching out and writing a detailed steps to follow.

I can confirm that it only works with AWS account ID for `aws-account-ids`. However, the CLI [docs](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/sqs/add-permission.html#options) for `aws-account-ids` parameter, it states

>The Amazon Web Services account numbers of the principals who are to receive permission.

which means it takes in account number(ID) instead of ARN.

The part you mentioned `The user, service, or account that receives permissions that are defined in a policy` means that the account ID you use for this operation has to have permissions that are defined in IAM policy. I think that's how I understand it. 

To summarize, this is expected behavior as the documentation states that it takes accountID and not ARN. Please let me know if you have any further questions.

**Comment by majsty** [comment]
Thanks for your response @aBurmeseDev. The current functionality is a breach of principle of least privilege for obvious reasons!

I suggest you make this restriction more explicit in the docs, perhaps as follows:
![image](https://user-images.githubusercontent.com/8297517/200980413-7a28de35-e238-437a-b35c-983eb19e8ba4.png)

And this ticket should now become a feature request. Should I raise a separate ticket or can we convert this one?

**Comment by aBurmeseDev** [comment]
@majsty - thank you for your feedback. I reached out to SQS team to confirm this behavior and asked for clarification. Since it's related to our Service API rather than AWS CLI, I'm going to transfer this issue to our cross-SDK repository and will follow up with updates there.
